app.go

```diff
@@ -461,6 +460,16 @@ func (app *App) StopTimeout() time.Duration {
 	return app.stopTimeout
 }
 
+func (app *App) dotGraph() DotGraph {
+	var b bytes.Buffer
+	msg := "digraph { graph []; error [shape=plaintext label=\"%q\"]; }"
+	if err := dig.Visualize(app.container, &b); err != nil {
```

Comment:
Should I just ignore this error and comment? I can't get past the test coverage requirement unless I test this and I haven't figured out a simple way to trigger the error from `dig.Visualize`.

Reply:
I don't see an easy way to trigger this error condition but we should still have logic to handle it. It'll have to remain uncovered.

---

If we *really* wanted to cover it, we could declare a private interface for the buffer and a private field for generating buffers.

```
type buffer interface {
  io.Writer
  fmt.Stringer
}

type App struct {
  newBuffer buffer
}
```

Then in the test you would return an implementation that returns an error on write.

Reply:
Thanks - I think we will need to cover this especially if we let the error trigger panic as Akshay suggested.